### PR TITLE
feat(Firma con IO): [SFEQS-1402] Generic error component refactoring with copy updates

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
This PR changes the generic error component by adding the retry prop and all locales for each error scenario.

## List of changes proposed in this pull request
- Update `GenericErrorComponent`
- Update all screens that make use of it
- Update related test

## How to test
With `io-dev-server-api` change fci config to generate FCI messages. Using `rejectedCount` to get a message with signature request REJECTED. The REJECTED status show a GenericErrorComponent with issuer email. In `FciRouterScreen` try to return e pot.error to show a GenericErrorComponent. (If not yet merged use this branch https://github.com/pagopa/io-dev-api-server/pull/216)

### CODE REVIEW

File: ts/features/fci/components/SuccessComponent.tsx [hunk 1]
```diff
@@ -25,9 +29,30 @@ const SuccessComponent = (props: {
   const now = new Date();
   const expires_at = new Date(props.signatureRequest.expires_at);
   const updated_at = new Date(props.signatureRequest.updated_at);
+  const issuer_email = props.signatureRequest.issuer.email;
   const status = props.signatureRequest.status;
   const dispatch = useIODispatch();
 
+  const publicKeyOption = useIOSelector(
+    createSelector(lollipopPublicKeySelector, V => V)
```
Comment (hevelius) [inline ts/features/fci/components/SuccessComponent.tsx:37]: @shadowsheep1 what do you think about this temp solution?

File: ts/features/fci/components/ErrorComponent.tsx [hunk 2]
```diff
@@ -1,41 +1,55 @@
...
+    bordered: true,
+    onPress: props.onPress,
+    title: I18n.t("global.buttons.close")
+  };
+  const retryButtonProps = {
+    block: true,
+    primary: true,
+    onPress: props.onPress,
+    title: I18n.t("global.buttons.retry")
+  };
```
Comment (shadowsheep1) [inline ts/features/fci/components/ErrorComponent.tsx:34]: What do you think writing something like that?

 ```ts
const buttonProps = {
    block: true,
    primary: true,
    onPress: props.onPress
  };

  const closeButtonProps = {
    ...buttonProps,
    bordered: true,
    title: I18n.t("global.buttons.close")
  };

  const retryButtonProps = {
    ...buttonProps,
    title: I18n.t("global.buttons.retry")
  };
```
We could improve code readability by separating between "standard" props and the one specific for "close" or "retry" buttons.

File: ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx [hunk 3]
```diff
@@ -134,9 +148,17 @@ export const FciDocumentPreviewScreen = (
...
         )}
       </SafeAreaView>
     </BaseScreenComponent>
+  ) : (
+    <GenericErrorComponent
+      title={I18n.t("features.fci.errors.generic.default.title")}
+      subTitle={I18n.t("features.fci.errors.generic.default.subTitle")}
+      retry={true}
+      onPress={() => dispatch(fciEndRequest())}
+    />
```
Comment (shadowsheep1) [inline ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx:162]: Ain't this case covered at lines 60-70?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx (hunk 3 above)
Comment (hevelius) [reply ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx:162]: Good catch!

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**